src/vm/excep.cpp

```diff
@@ -11536,28 +11418,7 @@ BOOL CEHelper::CanMethodHandleException(CorruptionSeverity severity, PTR_MethodD
 
     if (severity > NotCorrupting)
     {
-        // If the method lies in an assembly built for pre-V4 runtime, allow the runtime
-        // to look for exception handler for the CE.
-        BOOL fIsMethodInPreV4Assembly = FALSE;
-        fIsMethodInPreV4Assembly = CEHelper::IsMethodInPreV4Assembly(pMethodDesc);
-
-        if (!fIsMethodInPreV4Assembly)
-        {
-            // Method lies in an assembly built for V4 or later runtime.
-            LOG((LF_EH, LL_INFO100, "CEHelper::CanMethodHandleException - Method is in an assembly built for V4 or later runtime.\n"));
-
-            // Depending upon the corruption severity of the exception, see if the
-            // method supports handling that.
-            LOG((LF_EH, LL_INFO100, "CEHelper::CanMethodHandleException - Exception is corrupting.\n"));
-
-            // Check if the method can handle the severity specified in the exception object.
-            fLookForExceptionHandlersInMethod = CEHelper::CanMethodHandleCE(pMethodDesc, severity, fCalculateSecurityInfo);
-        }
-        else
-        {
-            // Method is in a Pre-V4 assembly - allow it to be examined for processing the CE
-            fLookForExceptionHandlersInMethod = TRUE;
-        }
+        fLookForExceptionHandlersInMethod = FALSE;
```

Comment:
This is not an appropriate change - not only we are incompletely removing the functionality but if we need to add additional policies for this feature, this will not be available.

The right fix would be to check for CorruptionSeverity against the ExceptionTracker/ExInfo, when it is setup early in the personality routine, and failfast if it is corrupting.